build-dotnet/build.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -e
+
+# Config
+
+dnf install -y clang python-unversioned-command
+
+git clone https://github.com/raulsntos/godot-dotnet
+cd godot-dotnet
+git checkout upgrade-assistant-plus-source-code-plugin-wip
+
+./build.sh --productBuild --warnAsError false /p:GenerateGodotBindings=true
```

Comment:
Should we build Godot to generate the API json and before building godot-dotnet?

Reply:
There is a small chance it could break godot-dotnet, so we probably don't want to do this in the script when building a release, we should do it _before_ building a release. We'd push the changes to the repo like godot-cpp (example: https://github.com/godotengine/godot-cpp/commit/e83fd0904c13356ed1d4c3d09f8bb9132bdc6b77).

The only difference is godot-cpp makes those commits after a release, and we probably want to do it before the release but after we tagged the commit we'll use for the release. So that we can publish the NuGet packages at the same time.

Reply:
What I'm saying is that sometimes the required changes can't be automated. For example:

- `dev3` added new API to GDExtension with a pointer type we don't recognize so I added code to exclude the API for now until we can properly support the type: https://github.com/raulsntos/godot-dotnet/commit/8d84f62365bda055476a40c26b405a66abdcf0dd
- `dev4` added a new field `hash_compatibility` that we weren't expecting, we intentionally want to fail on unknown fields so it can notify us when new fields are added. So I added the code to recognize the field: https://github.com/raulsntos/godot-dotnet/commit/4e9d54d07ceb321f96a7bab66752c23ca099bc24